pkg/virt-api/rest/subresource.go

```diff
@@ -517,6 +518,82 @@ func (app *SubresourceAPIApp) RestartVMRequestHandler(request *restful.Request,
 	response.WriteHeader(http.StatusAccepted)
 }
 
+func (app *SubresourceAPIApp) RenameVMRequestHandler(request *restful.Request, response *restful.Response) {
+	name := request.PathParameter("name")
+	namespace := request.PathParameter("namespace")
+
+	// Read the new name of the VM
+	newNameBuf, err := ioutil.ReadAll(request.Request.Body)
+
+	if err != nil {
+		writeError(errors.NewConflict(v1.Resource("virtualmachine"),
+			name, err), response)
+		return
+	}
+
+	newName := string(newNameBuf)
+
+	request.Request.Body.Close()
+
+	if name == newName {
+		writeError(errors.NewConflict(v1.Resource("virtualmachine"),
+			name, fmt.Errorf("The new name cannot be the same as the old name")), response)
+		return
+	}
+
+	// Make sure a VM with the newName doesn't exist
+	_, statusErr := app.fetchVirtualMachine(newName, namespace)
+
+	if statusErr == nil {
+		writeError(errors.NewConflict(v1.Resource("virtualmachine"),
+			name, fmt.Errorf("A VM with the new name already exists")), response)
+		return
+	} else if statusErr.ErrStatus.Code != http.StatusNotFound {
+		writeError(statusErr, response)
+		return
+	}
+
+	vm, statusErr := app.fetchVirtualMachine(name, namespace)
+
+	if statusErr != nil {
+		writeError(statusErr, response)
+		return
+	}
+
+	if vm.Spec.Running == nil {
+		writeError(errors.NewInternalError(fmt.Errorf("Could not determine the running status of the vm")),
+			response)
+		return
+	} else if *vm.Spec.Running {
+		writeError(errors.NewConflict(v1.Resource("virtualmachine"),
+			name, fmt.Errorf("Renaming a running vm is not allowed")), response)
+		return
+	}
+
+	// Check for renaming annotation on vm
+	if vm.Annotations != nil {
+		_, isScheduledForRenaming := vm.Annotations[v1.RenameToAnnotation]
+
+		if isScheduledForRenaming {
```

Comment:
I don't think it's relevant, because if a `renameTo` annotation exists, regardless of its value, virt-controller has already started handling the rename process and will create a new VM and delete the old one.
Say I checked if `vm.Annotations["renameTo"] == newName`, what would you suggest I do then?